app/content/profile/index.js

```diff
@@ -125,7 +138,68 @@ function Index() {
   }
 
   const handleImageClick = () => {
-    console.log("TODO: implement and open an image upload modal!");
+    setIsImageUploadModalVisible(true);
+  };
+
+  // Function to handle closing the image upload modal
+  const hideImageUploadModal = () => {
+    setIsImageUploadModalVisible(false);
+  };
+
+  const firebaseProfileImageUpload = async (uri) => {
+    try {
+      setImageUploading(true);
+      // Fetch the image data from the URI
+      const response = await fetch(uri);
+      const blob = await response.blob();
+
+      // Get a reference to the Firebase Storage instance
+      const storage = getStorage();
+
+      // Create a reference to the storage location where the image will be stored
+      const storageRef = ref(storage, userId);
+
+      // Upload the image to Firebase Storage
+      const snapshot = await uploadBytes(storageRef, blob);
+
+      console.log("Uploaded the image blob to the Firebase storage:", snapshot);
+
+      // Get the download URL for the uploaded image
+      const downloadURL = await getDownloadURL(snapshot.ref);
+      await updateDoc(doc(db, "teams", currentTeamId, "users", userId), {
+        pfp: downloadURL,
+      });
+
+      queryClient.invalidateQueries({ queryKey: ["user", { userId }] });
```

Comment:
I think you should invalidate the team index page cache too. The query key for the team index page is the same as this one, but without the userId argument: `queryKey: ["user"]`

I have a [hook in my leaderboard PR](https://github.com/Golf-Drill-Challenge-App/Golf-App/pull/161/files#diff-959abf0664d80899f422689d973b845b794154cb9f44851a4decfb4f1cbad77e) for invalidating multiple keys, in the meantime you could probably just call queryClient.inValidateQueries multiple times.

Screenshots (profile tab updates / invalidates cache on button press, team tab does not)
![image](https://github.com/Golf-Drill-Challenge-App/Golf-App/assets/82061589/52e83a20-a114-4010-b8cd-1e91f169bb19)
![image](https://github.com/Golf-Drill-Challenge-App/Golf-App/assets/82061589/a2f2f0d7-04c3-4c2d-a8a0-f9280183c5e0)

Reply:
Wait, don't these need the currentTeamId as one of the object items?

Reply:
> Wait, don't these need the currentTeamId as one of the object items?

I think useUserInfo hook (the only one called on team index page) has currentTeamId as an optional parameter (inside an object): https://github.com/Golf-Drill-Challenge-App/Golf-App/blob/layout/hooks/useUserInfo.js#L10

I think the useAttempt and useLeaderboard hooks require currentTeamId (it is not inside an object in query key).

Reply:
I just noticed this issue for leaderboard pfps: https://github.com/Golf-Drill-Challenge-App/Golf-App/issues/167

When that issue is fixed and the actual pfp of players are shown on leaderboard, probably need to invalidate leaderboard cache too (useLeaderboard hook).